### DISCUSSION TITLE
hub/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9878] [Docs]
# hub/

Gain insights into training and deploying your YOLOv5 and YOLOv8 models with Ultralytics HUB. Explore pre-trained models, templates and various integrations.

https://docs.ultralytics.com/hub/

**Comment by tdengs** [comment]
Hi, is it possible to run pytorch lite models on Ultralytics Hub?

**Reply by pderrenger** [reply]
Hey there! 🌟 At the moment, Ultralytics HUB mainly supports training and deploying YOLOv5 and YOLOv8 models. For deploying PyTorch Lite models, HUB currently focuses on formats like ONNX, CoreML, TensorFlow, and more as mentioned in our docs.

However, converting PyTorch models to TorchScript could potentially be run on mobile devices, utilizing the optimizations for mobile. Here's a quick example on converting a model to TorchScript:

```python
import torch
model = torch.load('your_model.pt')  # Load your trained model
model = model.eval()  # Set to evaluation mode
traced_model = torch.jit.trace(model, torch.rand(1, 3, 640, 640))  # Example input tensor
traced_model.save("model_torchscript.pt")
```

This script converts a PyTorch model to TorchScript, which then can be used with mobile applications or other platforms supporting TorchScript execution.

For deploying PyTorch Lite models specifically via HUB or running them effectively on mobile devices, it would be beneficial to keep an eye on updates and new features we plan to roll out. We're constantly working to expand our support and features based on our community's needs. 🚀

Don't hesitate to drop questions or suggestions; we're here to help! Happy coding!

**Comment by Osmiz** [comment]
is it possible to run custom models in Ultralytics hub app?

**Reply by pderrenger** [reply]
Yes, you can definitely run custom models on Ultralytics HUB! 🚀 After training your YOLOv5 or YOLOv8 models, you can easily deploy them for inference. Here's a brief overview:

1. **Train Your Model:** Use HUB to train your YOLOv5 or YOLOv8 model with your custom dataset.
2. **Export and Deploy:** Once trained, you can export your model to various formats (ONNX, CoreML, etc.) and deploy it according to your needs.

Remember, the HUB is designed to make the process intuitive and straightforward, so you can get your models up and running in no time! If you encounter any hiccups or have more detailed questions, feel free to reach out on our GitHub issues or join the Discord community for support. Happy modeling! 🌟

**Comment by ghost** [comment]
In ultralytics hub app, I created a project and trained a detection model with my custom data data set and checked the respective model in ultralytics mobile app it worked well , but when I exported in core ML format and try use it in Xcode the classes Labels  are missing. I cannot use it in ios app beacuse that missing class labels. I didn't use any  3rd party platform ,I just downloaded from ultralytics hub and the class labels are missing for custom models

**Reply by glenn-jocher** [reply]
@Charan2128 thank you for bringing this to our attention. It seems like the class labels are not being included in the CoreML export. Please ensure you are using the latest version of the Ultralytics HUB and the mobile app. If the issue persists, try re-exporting the model and verify the labels in the exported file. For further assistance, you can refer to the [Ultralytics HUB documentation](https://docs.ultralytics.com/hub/) or raise an issue on GitHub.
